src/components/TopBar.test.ts

```diff
@@ -0,0 +1,72 @@
+import { screen } from '@testing-library/react';
+import { h } from 'react-hyperscript-helpers';
+import TopBar from 'src/components/TopBar';
+import { AuthState, authStore } from 'src/libs/state';
+import { renderWithAppContexts as render } from 'src/testing/test-utils';
+
+type AjaxExports = typeof import('src/libs/ajax');
+jest.mock('src/libs/ajax', (): Partial<AjaxExports> => {
+  return {
+    Ajax: jest.fn(),
+  };
+});
+
+jest.mock('src/auth/auth');
+
+jest.mock('react-notifications-component', () => {
+  return {
+    Store: {
+      addNotification: jest.fn(),
+      removeNotification: jest.fn(),
+    },
+  };
+});
+
+type NavExports = typeof import('src/libs/nav');
+jest.mock(
+  'src/libs/nav',
+  (): NavExports => ({
+    ...jest.requireActual<NavExports>('src/libs/nav'),
+    getLink: jest.fn(),
+  })
+);
+
+const itDoesNotShowTheSignInButton = (newState: Partial<AuthState>) => {
```

Comment:
This is a somewhat unusual way to write these tests. Take a look at `it.each`, which is a more standard way to iterate though variations.

Example: https://github.com/DataBiosphere/terra-ui/blob/a9fe75a4cac8424376addf4c1a3b34ca42172c44/src/libs/utils.test.ts#L41